nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/GrokParser.java

```diff
@@ -0,0 +1,243 @@
+            .addValidator(StandardValidators.CHARACTER_SET_VALIDATOR)
+            .defaultValue("UTF-8")
+            .build();
+
+    public static final PropertyDescriptor MAX_BUFFER_SIZE = new PropertyDescriptor
+            .Builder().name("Maximum Buffer Size")
+            .description("Specifies the maximum amount of data to buffer (per file) in order to apply the Grok expressions. Files larger than the specified maximum will not be fully evaluated.")
+            .required(true)
+            .addValidator(StandardValidators.DATA_SIZE_VALIDATOR)
+            .addValidator(StandardValidators.createDataSizeBoundsValidator(0, Integer.MAX_VALUE))
+            .defaultValue("1 MB")
+            .build();
+
+    public static final Relationship REL_MATCH = new Relationship.Builder()
+            .name("matched")
+            .description("FlowFiles are routed to this relationship when the Grok Expression is successfully evaluated and the FlowFile is modified as a result")
+            .build();
+
+    public static final Relationship REL_NO_MATCH = new Relationship.Builder()
+            .name("unmatched")
+            .description("FlowFiles are routed to this relationship when no provided Grok Expression matches the content of the FlowFile")
+            .build();
+
+    private List<PropertyDescriptor> descriptors;
+
+    private Set<Relationship> relationships;
+
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    private Grok grok;
+    private byte[] buffer;
+
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        final List<PropertyDescriptor> descriptors = new ArrayList<PropertyDescriptor>();
+        descriptors.add(GROK_EXPRESSION);
+        descriptors.add(GROK_PATTERN_FILE);
+        descriptors.add(DESTINATION);
+        descriptors.add(CHARACTER_SET);
+        descriptors.add(MAX_BUFFER_SIZE);
+        this.descriptors = Collections.unmodifiableList(descriptors);
+
+        final Set<Relationship> relationships = new HashSet<Relationship>();
+        relationships.add(REL_MATCH);
+        relationships.add(REL_NO_MATCH);
+        this.relationships = Collections.unmodifiableSet(relationships);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return this.relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+
+        final int maxBufferSize = context.getProperty(MAX_BUFFER_SIZE).asDataSize(DataUnit.B).intValue();
+        buffer = new byte[maxBufferSize];
+
+        try{
+            grok = Grok.create(context.getProperty(GROK_PATTERN_FILE).getValue());
+            grok.compile(context.getProperty(GROK_EXPRESSION).getValue());
+        }catch (GrokException e){
+            getLogger().error("Failed to initialize ExtractGrok due to: ", e);
```

Comment:
If this is the case, the Processor will not behave properly. In fact, if the Grok.create() throws an Exception, the previous value will still be stored in the 'grok' member variable, which means that if the user changes the value, and we can't create a Grok for the new value, the processor will continue to use the old value. Instead, we should avoid catching this Exception and just declare the method to throw GrokException.